### ISSUE TITLE
want the ability to make beekeeper like sql editor

### ISSUE THREAD

**Issue by Hady-Eslam** [beekeeper-studio/beekeeper-studio#256]
i have mysql file that i write all my commands in it then make it exported to my database
can you make beekeeper have the ability to open these files like editor without making connection
as it is very hard to use an editor and beekeeper at the same time.
or make beekeeper have the ability to save the tabs in sql files

**Comment by rathboma** [comment]
Yes we need to add a way to do this for sure. For the time being I think you can drag/drop the sql file into the editor window of a tab to use it?

**Comment by Hady-Eslam** [comment]
OK I will try

**Comment by rathboma** [comment]
Strawman for how this would work:

1. Double clicking .sql files:
  - opens in most recent Beekeeper window
  - If not currently connected to a DB, asks you to connect before opening

2. Saving a query:
  - By default still save to the internal SQL store (database)
  - Add a dropdown on the save box for 'export to file'
  - If query was opened from a .sql file, auto-populate the file name in the export option

3. New Menu entries -> File [import, export] query from file

### Why keep in the database?

We've gone back and forth about saving .sql files natively, or storing them in our internal DB. We're sticking with the DB for a few reasons:
- Gives us the ability to provide a quick-open (ctrl-p or ctrl-o) search feature (like Slack / VSCode)
- Gives us the ability to analyze queries and databases for table usages, and that type of useful information
- Gives us the ability to provide query versioning (most folks don't use git at all for .sql files)

**Comment by Hady-Eslam** [comment]
is it implemented or will be implemented in the next update ??

**Comment by phillpafford** [comment]
I don't want to add any scope to this effort as I really do need to be able to open an external .sql file and save to external as well. While I do see the benefit of having them in the db this does present a few issues

- Hard to migrate to another computer
- Hard to share with others/team
- Hard to backup

Maybe a hybrid approach is to allow the save and opening of an external .sql file but have a default directory where these are opened from or saved to, then scan the files into to DB?

**Comment by rathboma** [comment]
Agreed that there needs to be a way to interact with .sql files.

We're a bit under the hammer right now with non-BK stuff (homeschooling etc), but we'll need to put some really serious thought into how to handle this.

We're planning on building a system for syncing a workspace, maybe that should be file-based. Not sure yet.

Thanks for trying the project! We're only a small (super part time) team, so we're going as fast as we can :-)

**Comment by anthonydb** [comment]
Chiming in here to say that ability to open, write to, and save and external file is a key part of my workflow. I'd vote against saving in a DB. Thanks!

**Comment by RgnKjnVA** [comment]
A year later, is there any progress on this?  I'm trying out BeeKeeper Studio and like a lot of what it offers but the inability to handle sql files, particularly not persisting the last used sql file on startup, is a perplexing omission imo.

**Comment by amaury-hanser** [comment]
Hello, is there any news on the feature ? 
It would be nice to be able to open sql files in Beekeeper.

**Comment by rathboma** [comment]
Not yet, I'll be working on this in 2022, but for now you can drag/drop them into the editor area as a workaround!

**Comment by davedgd** [comment]
> Not yet, I'll be working on this in 2022, but for now you can drag/drop them into the editor area as a workaround!

This is very helpful for importing/loading sql, but it would be really great to have something basic for doing the export as well (manually copy & pasting the syntax to another tool notwithstanding)!

**Comment by jcquarto** [comment]
wouldn't integration with a git repo accomplish this? git would handle the persistance and modification of the .sql files. as far as i can tell all anyone is asking for is the ability to easily run a sql file with Beekeeper -- which it already does. We just need a file browser to browse the cloned repo from inside Beekeeper.  p.s. DataGrip does this

**Comment by rathboma** [comment]
Hey! The 5.0 release does a lot of technical stuff that we needed to add features like this one.

Stay tuned!

**Comment by davedgd** [comment]
> Hey! The 5.0 release does a lot of technical stuff that we needed to add features like this one.
> 
> Stay tuned!

Awesome -- this is the top thing students in my class are confused by (i.e., how do I save/load my scripts).
